V2.0

jianghuaw opened the pull request:
Please see the messages for each commits.

Commit:
Put the convert_node_to_xenserver.sh to backgroud

Put this script(invoked in rc.local) to be run in backgroud, so
that it will allow us to login XS from console if this script
got stuck. It's helpful for trouble-shooting.

Commit:
Handle 0 < VCPUs-at-startup <= VCPUs-max

With this commit, it won't depends on the how many VCPUs the XVA
has.

Commit:
Use "kpartx -dvs" in pair of "kpartx -avs" in the same way

We need specify "-p p" in "kpartx -dvs", otherwise sometime it's not
able to determine the delimter in the mapper. Then "kpartx -dvs"
can't remove the mapper devices correctly. So you will see the
following command of "xe vbd-unplug" got stuck.

Commit:
Reboot domU from dom0 instead of "reboot" in domU.

Sometime the reboot running in DOMU will cause the ssh session
closed before it return. So it will cause this script exit
as we set -e in this script.

Commit:
Support Ubuntu Xenial in devstack domU

Commit:
Support larger disk.

BobBall commented on nodepool/scripts/install_xenserver.sh:
```diff
@@ -17,11 +17,9 @@
 # limitations under the License.
 
-XENSERVER_XVA_URL=${NODEPOOL_XENSERVER_XVA_URL:-http://downloads.vmd.citrix.com/OpenStack/xenapi-in-the-cloud-appliances/prod_ci}
+XENSERVER_XVA_URL=http://downloads.vmd.citrix.com/OpenStack/xenapi-in-the-cloud-appliances/Xenial-VM-by-XS6.5-20G.xva
```
Why remove the ability to override?

jianghuaw replied:
Ah, sorry. definitely should keep it.

BobBall commented on nodepool/scripts/prepare_node.sh:
```diff
@@ -108,12 +112,7 @@ wget https://git.openstack.org/cgit/openstack-infra/system-config/plain/install_
 # This workaround is to cheat the package upgrade to be under chroot env so
 # that it won't restart upstart/init.
-sudo cp -p /usr/bin/ischroot /usr/bin/ischroot.bak
-sudo sh -c 'echo "exit 0" > /usr/bin/ischroot'
```
Why do we not need this any more?  If this is removed, the comment should also be deleted.

jianghuaw replied:
it's not needed anymore with Xenial. Surely will update the comments.

BobBall commented on nodepool/scripts/xenserver_helper_initramfs_premount:
```diff
@@ -33,7 +33,7 @@ NUMBER_OF_BLOCKS=$(tune2fs -l /dev/xvda1 | grep "Block count" | tr -d " " | cut
 SIZE_OF_PARTITION=$(expr $NUMBER_OF_BLOCKS \* 8)
 
-sfdisk -d /dev/xvda | sed -e "s,[0-9]\{8\},$SIZE_OF_PARTITION,g" > /tmp/new_layout
+sfdisk -d /dev/xvda | sed -e "s/[0-9]\{8,9\}/$SIZE_OF_PARTITION/g" > /tmp/new_layout
```
Is 9 big enough?
Should we use [0-9]{8}[0-9]* so it's dynamic?

jianghuaw replied:
yeah. Agree:-)

BobBall commented on nodepool/scripts/prepare_node.sh:
```diff
@@ -23,6 +23,11 @@ THIN=${THIN:-true}
 GIT_BASE=${GIT_BASE:-git://git.openstack.org}
 
+# delete stack user if exists, in case the base XVA has this user already.
+if id stack ; then
```
Why?  Does the existance of a stack user cause issues?

jianghuaw replied:
yes. If the stack exists, it will fail the stack.sh which will try to create this user. If it failed creating this user, it will be topped in the middle.

BobBall commented on nodepool/scripts/install_devstack_dependencies.sh:
```diff
@@ -23,7 +23,8 @@ if [ -f /usr/bin/yum ]; then
 elif [ -f /usr/bin/apt-get ]; then
-    if [ "$(lsb_release -c -s)" = "precise" ]; then
+    UBUNTU_RELEASE=$(lsb_release -c -s)
+    if [ "$UBUNTU_RELEASE" = "precise" ]; then
```
Might as well drop precise?

jianghuaw replied:
sure.

BobBall reviewed (changes requested):
See in-line comments for requested changes

Commit:
Fix some issues

1. specify memory size for all test nodes
2. ensure the path of /root/.ssh
3. pass hostname to devstack domU
4. ensure the user of jenkins
5. add a sync before halt host just in case
